### PR TITLE
Update to latest CloudEvent SDK

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents.csproj [hunk 1]
```diff
@@ -14,6 +14,9 @@
     <Compile Include="$(AzureCoreSharedSources)TaskExtensions.cs" Link="Shared\%(RecursiveDir)\%(Filename)%(Extension)" />
     <PackageReference Include="Azure.Messaging.EventGrid" />
     <PackageReference Include="CloudNative.CloudEvents" />
-    <PackageReference Include="System.Text.Json" />
+    <PackageReference Include="CloudNative.CloudEvents.SystemTextJson" />
+<!--      This is required to prevent version conflict stemming from The CloudNative.CloudEvents.SystemTestJson depending on -->
```
Comment (pakrym) [inline sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents.csproj:18]: nit formatting.

**Review by pakrym — APPROVED** [review]

File: sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/README.md [hunk 2]
```diff
@@ -54,13 +54,11 @@ EventGridPublisherClient client = new EventGridPublisherClient(
...
-    new CloudEvent(
-        type: "record",
-        source: new Uri("http://www.contoso.com"))
-        {
-            Data = "data"
+    new CloudEvent {
+        Type = "record",
+        Source = new Uri("http://www.contoso.com"),
+        Data = "data"
         };
```
Comment (jsquire) [inline sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/README.md:61]:
```suggestion
    new CloudEvent 
    {
        Type = "record",
        Source = new Uri("http://www.contoso.com"),
        Data = "data"
    };
```

File: sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/tests/Samples/CloudNativeSamples.cs [hunk 3]
```diff
@@ -23,13 +23,11 @@ public async Task SendCloudNativeEvents()
...
-                new CloudEvent(
-                    type: "record",
-                    source: new Uri("http://www.contoso.com"))
-                    {
-                        Data = "data"
+                new CloudEvent {
+                    Type = "record",
+                    Source = new Uri("http://www.contoso.com"),
+                    Data = "data"
                     };
```
Comment (jsquire) [inline sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/tests/Samples/CloudNativeSamples.cs:30]:
```suggestion
                new CloudEvent 
                {
                    Type = "record",
                    Source = new Uri("http://www.contoso.com"),
                    Data = "data"
                };
```

File: sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs [hunk 4]
```diff
@@ -90,19 +92,19 @@ public static class EventGridPublisherClientExtensions
             writer.WriteStartArray();
             foreach (var cloudEvent in cloudEvents)
             {
-                var attributes = cloudEvent.GetAttributes();
+                List<string> populatedAttributes = cloudEvent.GetPopulatedAttributes().Select(pair => pair.Key.Name).ToList();
```
Comment (jsquire) [inline sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs:95]:
```suggestion
```

File: sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs [hunk 5]
```diff
@@ -90,19 +92,19 @@ public static class EventGridPublisherClientExtensions
...
                 if (activityId != null &&
-                        !attributes.ContainsKey(TraceParentHeaderName) &&
-                        !attributes.ContainsKey(TraceStateHeaderName))
+                    !populatedAttributes.Contains(TraceParentHeaderName) &&
+                    !populatedAttributes.Contains(TraceStateHeaderName))
```
Comment (jsquire) [inline sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs:98]:
```suggestion
                    !(cloudEvent.GetPopulatedAttributes().Any(attr => attr.Name == TraceParentHaderName || attr.Name == TraceStateHeaderName))
```

Not sure if you're interested only in an exact match or if doing a `string.Equals` for an ordinal case-insensitive comparison would be better...

**Review by jsquire — APPROVED** [review]

File: sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs (hunk 5 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs:98]: The attributes have to be lowercase so it shouldn't matter.

File: sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs [hunk 6]
```diff
@@ -90,19 +92,17 @@ public static class EventGridPublisherClientExtensions
             writer.WriteStartArray();
             foreach (var cloudEvent in cloudEvents)
             {
-                var attributes = cloudEvent.GetAttributes();
                 if (activityId != null &&
-                        !attributes.ContainsKey(TraceParentHeaderName) &&
-                        !attributes.ContainsKey(TraceStateHeaderName))
+                    !(cloudEvent.GetPopulatedAttributes().Any(attr => attr.Name == TraceParentHaderName || attr.Name == TraceStateHeaderName))
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Microsoft.Azure.Messaging.EventGrid.CloudNativeCloudEvents/src/EventGridPublisherClientExtensions.cs:96]:
```suggestion
                    !cloudEvent.GetPopulatedAttributes().Any(attr => attr.Name == TraceParentHaderName || attr.Name == TraceStateHeaderName))
```

**Pull request merged**